tests/omprog-output-capture.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+# This file is part of the rsyslog project, released under ASL 2.0
+
+# This test tests the 'output' setting of omprog when the feedback
+# feature is not used (confirmMessages=off).
+
+. $srcdir/diag.sh init
+generate_conf
+add_conf '
+module(load="../plugins/omprog/.libs/omprog")
+
+template(name="outfmt" type="string" string="%msg%\n")
+
+:msg, contains, "msgnum:" {
+    action(
+        type="omprog"
+	    binary=`echo $srcdir/testsuites/omprog-output-capture-bin.sh`
+        template="outfmt"
+        name="omprog_action"
+        output=`echo $RSYSLOG_OUT_LOG`
+        fileCreateMode="0644"  # default is 0600
+    )
+}
+'
+startup
+. $srcdir/diag.sh injectmsg 0 10
+shutdown_when_empty
+wait_shutdown
+
+expected_output="[stdout] Starting
+[stderr] Starting
+[stdout] Received msgnum:00000000:
+[stderr] Received msgnum:00000000:
+[stdout] Received msgnum:00000001:
+[stderr] Received msgnum:00000001:
+[stdout] Received msgnum:00000002:
+[stderr] Received msgnum:00000002:
+[stdout] Received msgnum:00000003:
+[stderr] Received msgnum:00000003:
+[stdout] Received msgnum:00000004:
+[stderr] Received msgnum:00000004:
+[stdout] Received msgnum:00000005:
+[stderr] Received msgnum:00000005:
+[stdout] Received msgnum:00000006:
+[stderr] Received msgnum:00000006:
+[stdout] Received msgnum:00000007:
+[stderr] Received msgnum:00000007:
+[stdout] Received msgnum:00000008:
+[stderr] Received msgnum:00000008:
+[stdout] Received msgnum:00000009:
+[stderr] Received msgnum:00000009:
+[stdout] Terminating normally
+[stderr] Terminating normally"
+
+written_output=$(<$RSYSLOG_OUT_LOG)
```

Comment:
there is also a cmp function (cmp_equal) inside the testbench plumbing.

Reply:
You meant 'content-cmp' I assume. Changed where applicable.

Also I have corrected a minor problem in this 'content-cmp' helper function (in diag.sh): when the expected content was multiline, it was not displayed correctly in case of failure (double quotes missing in echo). I have included this minor change in this same PR, if you don't mind.